lib/models/room.js

```diff
@@ -185,6 +191,18 @@ Room.prototype.getAvatarUrl = function(baseUrl, width, height, resizeMethod,
  };
 
  /**
+  * Get the implicit room name (i.e. what a given user would see if the
+  * room had no m.room.name)
+  * @param {string} userId The userId from whose perspective we want
+  * to calculate the implicit name
+  * @return {string} The implicit room name
+  */
+ Room.prototype.getImplicitRoomName = function(userId) {
```

Comment:
as per the PR: "add a Room.getImplicitRoomName so clients can know what a room would be named if it had no explicit m.room.name.".  The use case is maintaining the RoomSettings UI to show what a room would be called if you try to delete its name.  The only reason it takes a userId is because calculateRoomName does too, and the Room object doesn't appear to give you a way to get at the current js-sdk user's userId.

I thought 'implicitRoomName' - i.e. the implicit name a room would have if a user hasn't specified an explicit one was fine terminology personally.  Do you have a better suggestion?